File: rashida_bengali/week01/mta/js/mta.js
console.log('up and running!');

const subwaySystem = {
  'N': ['Times Square', '34th', '28th', '23rd', 'Union Square', '8th'],
  'L': ['8th', '6th', 'Union Square','3rd', '1st'],
  '6': ['Grand Central', '33rd', '28th', '23rd', 'Union Square', 'Astor Place']
};

console.log(subwaySystem);
console.log(subwaySystem['N']);
console.log(subwaySystem['L']);
console.log(subwaySystem['6']);

const planTrip = function (lineStart, startStation,  endStation) {
  const indexStart = subwaySystem[lineStart].indexOf(startStation);
    console.log (indexStart);
  // const indexEnd = subwaySystem[lineEnd].indexOf(endStation);
  //   console.log (indexEnd);
  const indexEnd = subwaySystem[lineStart].indexOf(endStation);
    console.log (indexEnd);
  let tripList = `Your journey will be through the following stops on the ${lineStart} line: `;
    if (indexStart < indexEnd) {
      for (let i=indexStart; i<=indexEnd; i++) {
          tripList += subwaySystem[lineStart][i] + ', ';
      }
    } else if (indexStart > indexEnd) {
      for (let i=indexStart; i>=indexEnd; i--) {
        tripList += subwaySystem[lineStart][i] + ', ';
      }
    }
    console.log(tripList.slice(0, -2) + '.');

  // for (let i=indexStart; i<subwaySystem[lineStart].length; i++) {
  //   console.log(subwaySystem[lineStart][i]);
  // }
};

planTrip('N', '34th', '8th');
planTrip('N', '8th', '28th');



const planTripNew = function (lineStart, startStation, lineEnd, endStation) {
  const indexStart = subwaySystem[lineStart].indexOf(startStation);
    console.log (indexStart);
  const indexEnd = subwaySystem[lineEnd].indexOf(endStation);
    console.log (indexEnd);
  const indexCommonOne = subwaySystem[lineStart].indexOf('Union Square');
    console.log (indexCommonOne);
  const indexCommonTwo = subwaySystem[lineEnd].indexOf('Union Square');
    console.log (indexCommonTwo);
    let tripListOne = `Your journey will be through the following stops on the ${lineStart} line: `;
    let totalStops = 0;
    if (indexStart < indexCommonOne) {
      for (let i=indexStart+1; i<=indexCommonOne; i++) {
          tripListOne += subwaySystem[lineStart][i] + ', ';
          totalStops++;
      }
    } else if (indexStart > indexCommonOne) {
      for (let i=indexStart-1; i>=indexCommonOne; i--) {
        tripListOne += subwaySystem[lineStart][i] + ', ';
        totalStops++;
      }
    }
    console.log(tripListOne.slice(0, -2) + '.');
    console.log(`Change at Union Square station.`);
    let tripListTwo = `Your journey will be through the following stops on the ${lineEnd} line: `;
    if (indexCommonTwo < indexEnd) {
      for (let i=indexCommonTwo+1; i<=indexEnd; i++) {
          tripListTwo += subwaySystem[lineEnd][i] + ', ';
          totalStops++;
      }
    } else if (indexCommonTwo > indexEnd) {
      for (let i=indexCommonTwo-1; i>=indexEnd; i--) {
          tripListTwo += subwaySystem[lineEnd][i] + ', ';
          totalStops++;
      }
    }
    console.log(tripListTwo.slice(0, -2) + '.');
    console.log(`${totalStops} stops in total.`);
};

planTripNew('N', '34th', 'L', '1st');
planTripNew('N', 'Times Square', '6', '33rd');
planTripNew('L', '8th', '6', '28th');



// Reverse Trip

// const planTripReverse = function (lineStart, startStation,  endStation) {
//   const indexStart = subwaySystem[lineStart].indexOf(startStation);
//     console.log (indexStart);
//     const indexEnd = subwaySystem[lineStart].indexOf(endStation);
//       console.log (indexEnd);
//   for (let i=indexStart; i>=indexEnd; i--) {
//     console.log(subwaySystem[lineStart][i]);
//   }
// };

// planTripReverse('N', 'Union Square', 'Times Square');

File: rashida_bengali/week04/Ruby/mta.rb

$subway_system = {
  :N => ['Times Square', '34th', '28th', '23rd', 'Union Square', '8th'],
  :L => ['8th', '6th', 'Union Square','3rd', '1st'],
   6 => ['Grand Central', '33rd', '28th', '23rd', 'Union Square', 'Astor Place']
}

puts $subway_system

def plan_trip(line_start, start_station, line_end, end_station)

  index_start = $subway_system[line_start].index(start_station)
  index_end = $subway_system[line_end].index(end_station)

  index_common_one = $subway_system[line_start].index('Union Square')
  index_common_two = $subway_system[line_end].index('Union Square');

  trip_list_one = "Your journey will be through the following stops on the #{line_start} line: "

  total_stops = 0

  if index_start < index_common_one
    i = index_start+1
    while i <= index_common_one do
      trip_list_one += $subway_system[line_start][i] + ', '
      i = i + 1 # where you may freely set i to any value
      total_stops += 1
    end
  elsif index_start > index_common_one
    i = index_start-1
    while i >= index_common_one do
      trip_list_one += $subway_system[line_start][i] + ', '
      i = i - 1 # where you may freely set i to any value
      total_stops += 1
    end
  end

  trip_list_one[-2] = '.'

  trip_list_two = "Your journey will be through the following stops on the #{line_end} line: "

  if index_common_two < index_end
    i = index_common_two+1
    while i <= index_end do
      trip_list_two += $subway_system[line_end][i] + ', '
      i = i + 1 # where you may freely set i to any value
      total_stops += 1
    end
  elsif index_common_two > index_end
    i = index_common_two-1
    while i >= index_end do
      trip_list_two += $subway_system[line_end][i] + ', '
      i = i - 1 # where you may freely set i to any value
      total_stops += 1
    end
  end

  trip_list_two[-2] = '.'
  journey = trip_list_one + "Change at Union Square station. " + trip_list_two
  stops = "#{total_stops} stops in total."
  return journey + stops
end

puts "#{plan_trip :N, '34th', :L, '1st'}"
